server/datastore/mysql/queries.go

```diff
@@ -253,3 +253,19 @@ func (ds *Datastore) loadPacksForQueries(ctx context.Context, queries []*fleet.Q
 
 	return nil
 }
+
+func (ds *Datastore) ObserverCanRunQuery(ctx context.Context, queryID uint) (*bool, error) {
+	sql := `
+		SELECT observer_can_run
+		FROM queries
+		WHERE id = (?)
+	`
+	rows := []*bool{}
+
+	err := sqlx.SelectContext(ctx, ds.reader, &rows, sql, queryID)
```

Comment:
You probably want `sqlx.GetContext` here since you're getting the first element. If you don't use that, you'll have to check that `len(rows) > 0`

Reply:
Also, you probably don't need a pointer to `bool` in this function, right? Can `observer_can_run` be null? If so, you could do

```
SELECT COALESCE(observer_can_run, false) FROM queries WHERE id = ?
```

And then that would simplify the usage later on when you call this.

Reply:
So in the case of an error, we'd return `false` along with the error?

Reply:
Right